3_visualize/src/visualize_data.R

```diff
@@ -0,0 +1,71 @@
+############## Visualize data  ######################
+
+project_output_dir <- '3_visualize/out'
+
+dir.create(project_output_dir)
+
+
+plot_results <- function() {
+png(file = file.path(project_output_dir, 'figure_1.png'), width = 8, height = 10, res = 200, units = 'in')
+par(omi = c(0,0,0.05,0.05), mai = c(1,1,0,0), las = 1, mgp = c(2,.5,0), cex = 1.5)
+
+plot(NA, NA, xlim = c(2, 1000), ylim = c(4.7, 0.75),
+     ylab = "Test RMSE (°C)", xlab = "Training temperature profiles (#)", log = 'x', axes = FALSE)
+
+n_profs <- c(2, 10, 50, 100, 500, 980)
+
+axis(1, at = c(-100, n_profs, 1e10), labels = c("", n_profs, ""), tck = -0.01)
+axis(2, at = seq(0,10), las = 1, tck = -0.01)
+
+# slight horizontal offsets so the markers don't overlap:
+offsets <- data.frame(pgdl = c(0.15, 0.5, 3, 7, 20, 30)) %>%
+  mutate(dl = -pgdl, pb = 0, n_prof = n_profs)
+
+
+for (mod in c('pb','dl','pgdl')){
+  mod_data <- filter(eval_data, model_type == mod)
+  mod_profiles <- unique(mod_data$n_prof)
+  for (mod_profile in mod_profiles){
+    d <- filter(mod_data, n_prof == mod_profile) %>% summarize(y0 = min(rmse), y1 = max(rmse), col = unique(col))
+    x_pos <- offsets %>% filter(n_prof == mod_profile) %>% pull(!!mod) + mod_profile
+    lines(c(x_pos, x_pos), c(d$y0, d$y1), col = d$col, lwd = 2.5)
+  }
+  d <- group_by(mod_data, n_prof) %>% summarize(y = mean(rmse), col = unique(col), pch = unique(pch)) %>%
+    rename(x = n_prof) %>% arrange(x)
+  
+  lines(d$x + tail(offsets[[mod]], nrow(d)), d$y, col = d$col[1], lty = 'dashed')
+  points(d$x + tail(offsets[[mod]], nrow(d)), d$y, pch = d$pch[1], col = d$col[1], bg = 'white', lwd = 2.5, cex = 1.5)
+  
+}
+
+points(2.2, 0.79, col = '#7570b3', pch = 23, bg = 'white', lwd = 2.5, cex = 1.5)
+text(2.3, 0.80, 'Process-Guided Deep Learning', pos = 4, cex = 1.1)
+
+points(2.2, 0.94, col = '#d95f02', pch = 22, bg = 'white', lwd = 2.5, cex = 1.5)
+text(2.3, 0.95, 'Deep Learning', pos = 4, cex = 1.1)
+
+points(2.2, 1.09, col = '#1b9e77', pch = 21, bg = 'white', lwd = 2.5, cex = 1.5)
+text(2.3, 1.1, 'Process-Based', pos = 4, cex = 1.1)
+
+dev.off()
+}
+
+
+diagnostic_log_file <- function() {
```

Comment:
I haven't yet said anything about this, but I disagree with your design decision to put the creation of a text file within a phase called "3_visualize". In my mind, a function like this would fit somewhat better in "2_process"...or if you had organized your phases differently (e.g., if this were a larger or different project), maybe in a phase called something like "4_summarize", "4_report", or similar.

Reply:
i copied this comment into my new review